experimental/codepointtrie/src/provider.rs

```diff
@@ -0,0 +1,58 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[cfg_attr(any(feature = "serde", test), derive(serde::Deserialize))]
+pub struct UnicodeEnumeratedProperty {
```

Comment:
Issue: Only "real" data structs (zero-copy structs used at runtime) should be exposed in the public `provider` module.  These structs look like the ones coming from the TOML files.  I suggest moving these definitions to the test module for now, and possibly moving them to a shared location you implement the transformer.